robot/impl/local_robot.go

```diff
@@ -861,6 +895,14 @@ func (r *localRobot) Reconfigure(ctx context.Context, newConfig *config.Config)
 	if r.revealSensitiveConfigDiffs {
 		r.logger.Debugf("(re)configuring with %+v", diff)
 	}
+
+	// Sync Packages before reconfiguring rest of robot.
+	// TODO: Make this non-blocking so other resources that do not require packages can run before package sync finishes.
+	err = r.packageManager.Sync(ctx, newConfig.Packages)
```

Comment:
I do wonder if we should run this even earlier. I guess we can always move this if we need to